builder error: cannot convert 'struct array' to 'struct array *'

### Describe the bug

A builder error is found when running the following code.

### Reproduction Steps

```v 
struct Balise {
}

fn escape_tag(mut parents []Balise) {
                dump(parents)
                stack := []&Balise{}
                dump(stack)
}
```

### Expected Behavior

No builder error.

### Current Behavior

```
==================
/tmp/v_1000/main.01J0K7CCEACS17T5B1Z0DCW7AH.tmp.c:13228: error: cannot convert 'struct array' to 'struct array *'
...
==================
(Use `v -cg` to print the entire error message)

builder error
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.6 294f7e4.bebe943

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.6 294f7e4.bebe943
OS: linux, Linux version 6.7.11-200.fc39.x86_64 (mockbuild@bdb8d24f206645e2bda44c1cf867612d) (gcc (GCC) 13.2.1 20240316 (Red Hat 13.2.1-7), GNU ld version 2.40-14.fc39) #1 SMP PREEMPT_DYNAMIC Wed Mar 27 16:50:39 UTC 2024
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/dumpexpr.v
+++ b/vlib/v/gen/c/dumpexpr.v
@@ -55,7 +55,7 @@ fn (mut g Gen) dump_expr(node ast.DumpExpr) {
 		name = name[3..]
 	}
 	dump_fn_name := '_v_dump_expr_${name}' +
-		(if expr_type.is_ptr() { '_ptr'.repeat(expr_type.nr_muls()) } else { '' })
+		(if expr_type.is_ptr() { '__ptr'.repeat(expr_type.nr_muls()) } else { '' })
 	g.write(' ${dump_fn_name}(${ctoslit(fpath)}, ${line}, ${sexpr}, ')
 	if expr_type.has_flag(.shared_f) {
 		g.write('&')
@@ -156,7 +156,7 @@ fn (mut g Gen) dump_expr_definitions() {
 			str_dumparg_ret_type = str_dumparg_type
 		}
 		dump_fn_name := '_v_dump_expr_${name}' +
-			(if is_ptr { '_ptr'.repeat(typ.nr_muls()) } else { '' })
+			(if is_ptr { '__ptr'.repeat(typ.nr_muls()) } else { '' })
 
 		// protect against duplicate declarations:
 		if dump_already_generated_fns[dump_fn_name] {

--- a/vlib/v/slow_tests/inout/dump_array_of_ref.out
+++ b/vlib/v/slow_tests/inout/dump_array_of_ref.out
@@ -1,0 +1,10 @@
+[vlib/v/slow_tests/inout/dump_array_of_ref.vv:6] parents: &[Balise{
+    a: 11
+}, Balise{
+    a: 22
+}]
+[vlib/v/slow_tests/inout/dump_array_of_ref.vv:10] stack: [&Balise{
+    a: 10
+}, &Balise{
+    a: 20
+}]

--- a/vlib/v/slow_tests/inout/dump_array_of_ref.vv
+++ b/vlib/v/slow_tests/inout/dump_array_of_ref.vv
@@ -1,0 +1,16 @@
+struct Balise {
+	a int
+}
+
+fn escape_tag(mut parents []Balise) {
+	dump(parents)
+	mut stack := []&Balise{}
+	stack << &Balise{10}
+	stack << &Balise{20}
+	dump(stack)
+}
+
+fn main() {
+	mut bal := [Balise{11}, Balise{22}]
+	escape_tag(mut bal)
+}

